### ISSUE TITLE
Deleted repos in github still showing up in sourcegraph results

### ISSUE THREAD

**Issue by superhsu** [sourcegraph/sourcegraph#35958]
- **Sourcegraph version:** <3.38.1>
- **Platform information:** <Docker-compose>

#### Steps to reproduce:

1. Delete repositories/old code in GitHub and search for it within Sourcegraph

#### Expected behavior:

Deleted repositories should not show up in Sourcegraph search results

#### Actual behavior:

Search results for the repo are listed in Sourcegraph, but upon navigating to the repo it 404s as it was deleted.

full context:
"Deleted repos are not being removed from the Sourcegraph index Seeing that the archival workflow was not working I started encouraging people to delete extremely old code. However, even this did not cause the repositories to be deindexed. Internal example search, internal repo link. You can see that results for the repo are listed in Sourcegraph, but upon navigating to the repo it 404’s as it was deleted. (Opinion) Individual user repository spaces should not be indexed"

**Comment by sourcegraph-bot-2** [comment]
Heads up @jplahn - the "team/repo-management" label was applied to this issue.

**Comment by varsanojidan** [comment]
@sashaostrikov and I have determined that cause for the repo deletions not being reflected in Sourceegraph was due to the fact that the external_service syncer process got killed and was not running since around May 18th, as evidenced by the data in the external_service_sync_jobs table.

**After a repo-updater restart, the syncer process started up again and began removing the problematic deleted repos. The customer's environment is looking good now, and the syncs are happening as expected.**

We plan to hop on a call  with the customer to try to find the cause of the syncer process crashing in the first place. current belief is that it was either an OOM or upgrade error, but still need to investigate.

We will also look into what the alerting is like around this process and create follow-up issues where we see a lack of observability/alerting.
